Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift

```diff
@@ -381,3 +428,33 @@ extension PortfolioStore: BraveWalletBraveWalletServiceObserver {
   public func onResetWallet() {
   }
 }
+
+extension PortfolioStore: PreferencesObserver {
+  func saveFilters(_ filters: Filters) {
+    isSavingFilters = true
+    defer {
+      isSavingFilters = false
+      update()
+    }
+    Preferences.Wallet.sortOrderFilter.value = filters.sortOrder.rawValue
+    Preferences.Wallet.isHidingSmallBalancesFilter.value = filters.isHidingSmallBalances
+    Preferences.Wallet.nonSelectedAccountsFilter.value = filters.accounts
+      .filter({ !$0.isSelected })
+      .map(\.model.address)
+    Preferences.Wallet.nonSelectedNetworksFilter.value = filters.networks
+      .filter({ !$0.isSelected })
+      .map(\.model.chainId)
+  }
+  public func preferencesDidChange(for key: String) {
+    guard !isSavingFilters else { return }
+    update()
+  }
+}
+
+extension Array {
+  /// `filter` helper that skips iterating through the entire array when not applying any filtering.
+  @inlinable public func optionallyFilter(shouldFilter: Bool, isIncluded: (Element) throws -> Bool) rethrows -> [Element] {
+    guard shouldFilter else { return self }
+    return try filter(isIncluded)
+  }
+}
```

Comment:
is this PR only implement the filter in Portfolio and the NFT one will be in another PR?

Reply:
d31189d50fced292c429e460cf15045fcca584c5